BDArmory/Control/MissileFire.cs

```diff
@@ -2934,7 +2948,15 @@ private void UpdateSelectedWeaponState()
                 //Debug.Log("[BDArmory.MissileFire]: =====selected weapon: " + selectedWeapon.GetPart().name);
                 if (!CurrentMissile || CurrentMissile.part.name != selectedWeapon.GetPart().name)
                 {
-                    CurrentMissile = selectedWeapon.GetPart().FindModuleImplementing<MissileBase>();
+                    using (var Missile = VesselModuleRegistry.GetModules<MissileBase>(vessel).GetEnumerator())
+                        while (Missile.MoveNext())
+                        {
+                            if (Missile.Current == null) continue;
+                            if (Missile.Current.part.name != selectedWeapon.GetPart().name) continue;
```

Comment:
`UnityEngine.Object.name` causes GC allocation for the returned string from the `.name` getter, so it's better to get the name for the selected weapon once outside the loop when it's first needed.
Even better would be to cache the weapon's part name as a field in `MissileBase` when it's first loaded and use that instead if it's going to be used frequently.

Reply:
Good point. I've gone through and fixed the other instances of part.name being used in MissileFire as well